app/src/forms/form/service.js

```diff
@@ -247,6 +286,7 @@ const service = {
     let trx;
     try {
       const obj = await service.readForm(formId);
+      service._validateAssigneeVisibilitySettings(data, obj);
```

Comment:
see note above about changing _validateAssigneeVisibilitySettings.
Also, make sure you update the test cases for create and update and add a test for whatever you name the function that will set showAssigneedInSubmissionsTable